frontend/src/app/projects/models/create-coworking-method/create-coworking-method.component.html

```diff
@@ -8,14 +8,93 @@
   class="wrapper flex-center"
 >
   <mat-card class="section-card">
-    <form [formGroup]="form">
-      <div formGroupName="credentials">
+    <h2>
+      Add a primary source for the model “{{ modelService.model.name }}” of the
+      project “{{ projectService.project.name }}”.
+    </h2>
+    <form [formGroup]="form" (submit)="onSubmit()">
+      <div formGroupName="urlGroup">
+        <fieldset *ngIf="availableGitSettings.length">
+          <mat-form-field appearance="fill">
+            <mat-label>Base integration URL</mat-label>
+            <mat-select
+              matInput
+              formControlName="baseUrl"
+              (selectionChange)="onSelect($event.value)"
+              [value]="availableGitSettings"
+            >
+              <mat-option
+                *ngFor="let gitSetting of availableGitSettings"
+                [value]="gitSetting"
+                >{{ gitSetting.url }}</mat-option
+              >
+            </mat-select>
+          </mat-form-field>
+        </fieldset>
         <fieldset>
           <mat-form-field appearance="fill">
-            <mat-label>Repository url</mat-label>
-            <input matInput formControlName="path" />
+            <mat-label
+              >Absolute URL
+              {{
+                availableGitSettings.length ? "or Relative URL" : ""
+              }}</mat-label
+            >
+            <input
+              matInput
+              (ngModelChange)="onInputChange($event)"
+              formControlName="inputUrl"
+            />
+            <mat-error
+              *ngIf="
+                form.controls.urlGroup.controls.inputUrl.errors
+                  ?.urlPrefixError &&
+                form.controls.urlGroup.controls.inputUrl.value
```

Comment:
The check is used here because an empty value has no valid prefix and thus the url prefix error is present. However, it does not make sense to print the error in this case, especially because the input field is not required.